File: BackEnd/backend/src/main/java/com/fitapp/backend/application/dto/routine/RoutineResponse.java
package com.fitapp.backend.application.dto.routine;

import lombok.Data;
import java.time.LocalDateTime;
import java.util.List;
import java.util.Set;

import com.fitapp.backend.infrastructure.persistence.entity.enums.DayOfWeek;

@Data
public class RoutineResponse {
    private Long id;
    private String name;
    private String description;
    private Long sportId;
    private String sportName;
    private Boolean isActive;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private List<RoutineExerciseResponse> exercises;
    private Set<DayOfWeek> trainingDays;
    private String goal;
    private Integer sessionsPerWeek;
}

File: FrontEnd/app/src/main/java/com/fitapp/appfit/response/routine/RoutineResponse.kt
package com.fitapp.appfit.response.routine

import java.time.LocalDateTime

data class RoutineResponse(
    val id: Long,
    val name: String,
    val description: String?,
    val sportId: Long?,
    val sportName: String?,
    val isActive: Boolean,
    val estimatedDuration: Int,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime,
    val exercises: List<RoutineExerciseResponse>
)